pilot/pkg/config/kube/ior/route.go

```diff
@@ -627,3 +367,155 @@ func hostHash(name string) string {
 	hash := sha256.Sum256([]byte(name))
 	return hex.EncodeToString(hash[:8])
 }
+
+func (r *route) reconcileGateway(config *config.Config, routes *v1.RouteList) error {
+	gateway, ok := config.Spec.(*networking.Gateway)
+
+	if !ok {
+		return fmt.Errorf("could not decode spec as Gateway from %v", config)
+	}
+
+	var serviceNamespace string
+	var serviceName string
+	var err error
+
+	serviceNamespace, serviceName, err = r.findService(gateway)
+
+	if err != nil {
+		return errors.Wrapf(err, "gateway %s/%s does not specify a valid service", config.Namespace, config.Name)
+	}
+
+	routeMap := make(map[string]v1.Route)
+
+	for _, v := range routes.Items {
+		routeMap[v.Name] = v
+	}
+
+	var result *multierror.Error
+
+	for _, server := range gateway.Servers {
+		for _, host := range server.Hosts {
+			var route *v1.Route
+			var err error
+
+			name := getRouteName(config.Namespace, config.Name, host)
+
+			_, found := routeMap[name]
+
+			if found {
+				route, err = r.updateRoute(config.Meta, host, server.Tls, serviceNamespace, serviceName)
+			} else {
+				route, err = r.createRoute(config.Meta, host, server.Tls, serviceNamespace, serviceName)
+			}
+
+			if err != nil {
+				result = multierror.Append(result, err)
+			} else {
+				delete(routeMap, route.Name)
+			}
+		}
+	}
+
+	for k, v := range routeMap {
+		IORLog.Debugf("clean up route %s for host %s", k, getHost(v))
+		if err := r.deleteRoute(&v); err != nil {
+			result = multierror.Append(result, err)
+		}
+	}
+
+	return result.ErrorOrNil()
+}
+
+func (r *route) processEvent(old, curr *config.Config, event model.Event) error {
+	if IORLog.GetOutputLevel() >= log.DebugLevel {
+		debugMessage := fmt.Sprintf("event %v arrived:", event)
+		if event == model.EventUpdate {
+			debugMessage += fmt.Sprintf("\told object: %v", old)
+		}
+		debugMessage += fmt.Sprintf("\tnew object: %v", curr)
+
+		IORLog.Debug(debugMessage)
+	}
+
+	var (
+		err       error
+		isManaged bool
+	)
+
+	isManaged, err = isManagedByIOR(*curr)
```

Comment:
```suggestion
	isManaged, err := isManagedByIOR(*curr)
```

Reply:
The reason I used this syntax is the `err` variable will be re-used. For example the code below, the `item, err := processList(list)` will not be happy about the use of  `err`.

```go
	list, err := findX()
	if err != nil {
		return err
	}

	items, err := processList(list)
	
	if err != nil {
		return err
	}

	return items
```

Reply:
Well, yes, but the variable isn't being shadowed in this part of the code, but further down. Whether you declare the variable using `var` or `:=` here makes no difference in the code further down - the `err` variable is shadowed in both cases.